src/mlpack/methods/ann/layer/noisylinear_impl.hpp

```diff
@@ -48,6 +48,50 @@ NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
   biasEpsilon.set_size(outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
+    NoisyLinear&& layer) :
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    weights(std::move(layer.weights))
+{
+  layer.inSize = 0;
+  layer.outSize = 0;
+  layer.weights = nullptr;
+  Reset();
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear const& layer)
+{
+  if( this != &layer)
+  {
+    inSize = layer.inSize;
+    outSize = layer.outSize;
+    weights = layer.weights;
+    Reset();
+  }
+  return *this;
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear&& layer)
+{
+  if (this != &layer)
+  {
+    inSize = std::move(layer.inSize);
+    layer.inSize = 0;
+    outSize = std::move(layer.outSize);
+    layer.outSize = 0;
+    weights = std::move(layer.weights);
+    layer.weights = nullptr;
```

Comment:
Same comment as earlier here---I don't think we want to set this to `nullptr`.  In fact, I'm surprised that this compiles at all; maybe there is something I overlooked?

Reply:
Yeah, I see! I think `nullptr` evaluates to a `prvalue`, and therefore can still (wrongly) be set in the right hand side? And also, the test still works because it does not care about what happens with the original layer, all it does is to check whether the moved network (containing moved layer) 's prediction is the same as that the original network.
I will make change to this. Thanks for the review!

Reply:
Interesting.  Yeah, in the test, it's not clear how to test that the network is fully and correctly uninitialized.  So I'm not sure that it's worth figuring out what to do there. :+1: